### DISCUSSION TITLE
Websocket TLS support?

### DISCUSSION THREAD

**Discussion by hanusek** [wasm-bindgen/wasm-bindgen#3438] [Q&A]
Hello.

I can't find information about support for TLS encryption in your [docs](https://rustwasm.github.io/wasm-bindgen/examples/websockets.html).
Does your [WebSocket](https://docs.rs/web-sys/latest/web_sys/struct.WebSocket.html) support TLS?

Please help.

**Comment by Liamolucko** [comment]
`web_sys::WebSocket` is just a binding to [the `WebSocket` API](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket) on the web. It uses TLS when you start the URL to connect to with `wss://` instead of `ws://`, the same as `https:` vs `http:`.

**Comment by hanusek** [comment]
@Liamolucko I can't connect. 
My server log:
```log
2023-03-28T10:36:36.361520Z  INFO auth: websocket - accepted connection
2023-03-28T10:36:36.363409Z DEBUG rustls::server::hs: decided upon suite TLS13_AES_128_GCM_SHA256    
2023-03-28T10:36:36.364312Z  WARN rustls::conn: Sending fatal alert HandshakeFailure    
Error: Custom { kind: InvalidData, error: PeerIncompatibleError("no overlapping sigschemes") }
```
How to set TLS/SSL protocol context?
How to use cert?
I guess I should use [new_with_str](https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.WebSocket.html#method.new_with_str)

In python I can set it:
```        
ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
ssl_context.load_verify_locations(pathlib.PurePath("../keys/eddsa-ca.cert"))
ssl_context.check_hostname = False
websockets.connect("wss://localhost:9999", ssl=ssl_context)
```

**Comment by Liamolucko** [comment]
The `WebSocket` API doesn't have any way of setting TLS options as far as I'm aware, that's completely decided by the browser.

`new_with_str` isn't what you're looking for, that sets the [WebSocket subprotocol](https://developer.mozilla.org/en-US/docs/Web/HTTP/Protocol_upgrade_mechanism#sec-websocket-protocol), which has nothing to do with TLS.
